src/pages/instances/forms/InstanceRename.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC, useState } from "react";
+import { LxdInstance } from "types/instance";
+import { Button, Input } from "@canonical/react-components";
+import { renameInstance } from "api/instances";
+import { useNotify } from "context/notify";
+import { useNavigate } from "react-router-dom";
+import SubmitButton from "components/SubmitButton";
+
+interface Props {
+  instance: LxdInstance;
+  project: string;
+  closeForm: () => void;
+}
+
+const InstanceRename: FC<Props> = ({ instance, project, closeForm }) => {
+  const navigate = useNavigate();
+  const notify = useNotify();
+  const [name, setName] = useState(instance.name);
+  const [isSubmitting, setSubmitting] = useState(false);
+
+  const handleSubmit = () => {
+    if (instance.name === name) {
+      closeForm();
+      return;
+    }
+    setSubmitting(true);
+    renameInstance(instance.name, name, project)
+      .then(() => {
+        navigate(
+          `/ui/${project}/instances/detail/${name}`,
+          notify.queue(notify.success("Instance renamed."))
+        );
+        closeForm();
+      })
+      .catch((e) => notify.failure("", e, undefined, "Renaming failed"))
+      .finally(() => setSubmitting(false));
+  };
+
+  return (
+    <>
+      <Input
+        autoFocus
+        className="u-no-margin--bottom name-input"
+        onChange={(e) => setName(e.target.value)}
+        onKeyUp={(e) => e.key === "Enter" && handleSubmit()}
+        type="text"
+        value={name}
+      />{" "}
```

Comment:
What is the purpose of the `{" "}` here?

Reply:
Wired, who put the space in there? oO

Reply:
:eyes: 
Thanks for removing it from the `ProfileRename`. It's still there for `InstanceRename` though.